### PR TITLE
A GSForm wrapper for material-ui/AutoComplete

### PR DESCRIPTION
**Pull request by lperson**
## Description

This PR introduces a component that wraps material-ui/AutoComplete so it can be included in a GSForm.  For example:

```
          <GSForm
            schema={formSchema}
            value={{ colors: colors[1] }}
            onChange={() => {}}
          >
            <Form.Field
              name="colors"
              type="autocomplete"
              choices={colors}
              hintText="What's your favorite color?"
            />
```

I intend to use this in the PR for syncing Spoke answers to canvass results in VAN.

# Checklist:

- [ ] I have manually tested my changes on desktop and mobile
- [x] The test suite passes locally with my changes
- [ ] If my change is a UI change, I have attached a screenshot to the description section of this pull request
- [x] [My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#your-first-code-contribution), or has a documented reason in the description why it’s longer
- [ ] I have made any necessary changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] My PR is labeled [WIP] if it is in progress

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -22,7 +23,8 @@ Form.addInputTypes({
   email: GSTextField,
   script: GSScriptField,
   select: GSSelectField,
-  password: GSPasswordField
+  password: GSPasswordField,
+  autocomplete: GSAutoComplete
```
Comment (schuyler1d) [inline src/components/App.jsx:27]: Is it possible to do this without including it in input types here?  (instead, using it by directly importing `GSAutoComplete`)?

My understanding is that addInputTypes is deprecated in newer React-formal versions -- so the others here will have to be changed the same way, but it would be good not to add to the migration burden.

See https://github.com/jquense/react-formal/issues/167

File: package.json [hunk 2]
```diff
@@ -172,7 +172,7 @@
     "json2csv": "^3.6.2",
     "lint-staged": "^9.2.0",
     "mockdate": "^2.0.2",
-    "nodemon": "^1.9.2",
+    "nodemon": "^2.0.2",
```
Comment (schuyler1d) [inline package.json]: is this related to this change?  If not, let's not mix things together.

File: src/components/App.jsx (hunk 1 above)
Comment (schuyler1d) [reply src/components/App.jsx:27]: @lperson pointed out to me that until we upgrade react-formal, we can't do it any other way -- so it's chicken/egg, and we shouldn't block on this.

Comment (lperson) [reply src/components/App.jsx:27]: In light of this, would you be so kind as to approve 🙏

**Comment by lperson** [comment]
@ibrand why did you close this without merging?

**Comment by ibrand** [comment]
That's weird... certainly didn't mean to close it! Thanks for seeing that. Reopening! Got in a weird state while making stage-main just now

**Comment by lperson** [comment]
@schuyler1d and @ibrand would it be possible to merge this into stage main.  I'm using it for the action handler work and it would be useful to have it not show up in the diffs for that.  Thank you!

**Comment by schuyler1d** [comment]
I think this PR still has yarn.lock churn that's unrelated to this change.

If `main` is merged into this, I think there's already a twilio upgrade there, so hopefully it will merge ok, and then there won't be a diff.

That might be ok anyway, but please please please -- let's keep these PRs contained to the needed changes.  There's a lot of community activity and we need to stay in our own focused areas.
